dodano funkcje parsowania i sortowania listy pracowników po pensji

JoAnna578 opened the pull request:
Dodano plik main.js z podstawową strukturą pod implementację funkcji do sortowania listy pracowników.

JoAnna578 committed:
dodano funkcje parsowania i sortowania listy pracowników po pensji

Dodano plik main.js z podstawową strukturą pod implementację funkcji do sortowania listy pracowników.

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1,3 +1,32 @@
+  if (!list) return; // guard, jeśli lista nie istnieje
+
+  // Funkcja sortująca elementy listy po pensji malejąco
+  function sortBySalaryDesc(listElement) {
```
Well done on implementing the sorting logic. According to the task description, this function should be named `sortList`. Please rename it to match the requirement.

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1,3 +1,32 @@
+  }
+
+  // Funkcja zwracająca tablicę obiektów
+  function getEmployeesArray(listElement) {
```
This function correctly creates the array of employees. However, the task requires this function to be named `getEmployees`. Please update the name to meet the requirement.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Great job on implementing the core logic for sorting and parsing the employee data! The functionality is correct and works as expected. However, a change is needed to fully meet the task requirements. The function names used in your script do not match the ones specified in the task description. Please rename `sortBySalaryDesc` to `sortList` and `getEmployeesArray` to `getEmployees`. After you make these adjustments, your solution will be perfect. Keep up the great work!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

JoAnna578 committed:
poprawa kodu zgodnie z testami

Usunięto nieużywaną zmienną, zachowano pełne linie bez łamania,
aby testy lintujące i funkcjonalne przeszły poprawnie.

JoAnna578 committed:
Sortowanie listy i tworzenie tablicy pracowników

Dodano funkcje:
- sortBySalary: sortuje elementy listy po pensji malejąco
- getEmployeesArray: tworzy tablicę obiektów pracowników
Linie krótsze niż 80 znaków, brak console.log, zgodne z linterem.

JoAnna578 committed:
Sortowanie listy i generowanie tablicy pracowników

JoAnna578 committed:
Poprawa funkcji i globalnego employees dla testów

JoAnna578 committed:
Poprawa sortowania

JoAnna578 committed:
Poprawa zgodnie z HTML

JoAnna578 committed:
Koncowa poprawa

JoAnna578 committed:
Kolejna poprawa

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1,3 +1,42 @@
+}
+
+// Funkcja sortująca elementy po pensji malejąco
+function sortBySalary(items) {
```
This function is required to be named `sortList`. According to the requirements, this function should also accept the `<ul>` list element as an argument and handle appending the sorted items back into it, rather than just returning a sorted array.

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1,3 +1,42 @@
+}
+
+// Funkcja tworząca tablicę obiektów pracowników
+function getEmployeesArray(items) {
```
This function should be named `getEmployees` as specified in the task description. The requirements also state that it should accept the `<ul>` list element as its argument.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Thank you for your submission. The core logic for parsing and sorting the employee data is working well. However, this submission requires changes because it does not yet meet the specific technical requirements for function naming and structure, which were also mentioned in the last review.

Specifically, please make the following adjustments:

1.  **Rename your functions.** The task requires the functions to be named `sortList` and `getEmployees`. Currently, they are named `sortBySalary` and `getEmployeesArray`.
2.  **Update the function signatures.** Both functions must be updated to accept the `<ul>` list element as their only argument.
3.  **Adjust function responsibility.** The `sortList` function should handle appending the sorted list items back into the DOM. This logic is currently happening in the global scope and should be moved inside the function.

Addressing these points will ensure your solution aligns with the project's requirements.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

JoAnna578 committed:
Wprowadzone sugestie AI mentora

JoAnna578 committed:
Poprawa Run

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Excellent work on this update! You have successfully addressed all the points from the previous review, and the code now fully meets all task requirements. The functions are correctly named, accept the proper arguments, and the logic for sorting and data parsing is implemented perfectly. Your solution is clean, correct, and complete, so I am approving it. Great job!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

JoAnna578 committed:
Nowy kod funkcji

JoAnna578 committed:
Mała poprawka

JoAnna578 committed:
Małe poprawki 2

JoAnna578 committed:
Run run run

JoAnna578 committed:
Run run
